ligo/stablecoin/actions.ligo

```diff
@@ -52,7 +142,96 @@ function debit_from
       }
   | None -> failwith ("Given address is not present in ledger")
   end
-} with store with record [ ledger = updated_ledger ]
+} with store with record
+    [ ledger = updated_ledger
+    ; total_supply = abs (store.total_supply - parameter.amount) // TODO: possible bug here
+    ]
+
+(*
+ * Converts a transfer item into a safelist transfer item
+ *)
+function convert_to_safelist_transfer
+  (const tp : transfer_param
+  ) : safelist_transfer_item is block
+  { const sl_transfer_item : safelist_transfer_item =
+      (record
```

Comment:
The type of the record can be inherited here, no need for an additional annotation and so it can be refactored without using `block` construction at all

Reply:
Changed.